docs/gitbook/architecture/permission_model.adoc

```diff
@@ -0,0 +1,172 @@
+with Interbit. The following discussion provides a conceptual framework
+for understanding how Interbit manages permissions. Follow the cross
+reference links for implementation details.
+
+
+== Chain Access
+
+Interbit uses permissioned blockchains; all access to a blockchain is
+controlled with public key cryptography. If you hold a private key that
+corresponds to a public key added to an Interbit blockchain, you can
+access that blockchain. If not, you have no access.
+
+[WARNING]
+=========
+If you are the last user on a blockchain that has the permission to
+add or remove public keys, do not remove your own public key from the
+blockchain. If you do remove your own public key, **your access to that
+blockchain is permanently removed**.
+=========
+
+
+== Access Control List (ACL)
+
+Interbit uses a role-based access control list, or ACL, to manage
+permissions on a blockchain. An ACL uses the concepts of _roles_ and
+_permissions_ (also known as privileges, or capabilities) to describe
+the specific blockchain actions that each user, or group of users, is
+permitted to dispatch.
+
+
+=== Roles
+
+In an organization, each person has a job-based function, or _role_. For
+example, developers, senior developers, managers, etc. Each person is
+permitted to, or prohibited from, dispatching specific actions according
+to their role in the organization. Some roles apply to many people, and
+some roles apply to specific people.
+
+Roles work the same way for Interbit blockchains. Your application built
+on Interbit has various actions that can be dispatched to a
+blockchain. Some actions may be common to most users with access to
+the blockchain, others may be restricted to specific individuals.
+
+For example, in a voting application you might create a `can_vote` role, which
+permits users with that role to add their vote to the blockchain and to
+see voting results, and an `auditor` role, which permits users with that
+role to see all of the individual votes but not to vote.
+
+Each Interbit blockchain has a `root` role, which permits users with
+that role to dispatch any action. Typically, the creator of a
+blockchain would have the `root` role, but each blockchain can have
+custom roles assigned at creation.
+
+Users can be assigned to multiple roles. Assignment simply involves
+adding a user's public key to the role declaration. A blockchain's
+identifier can be added to a role, to permit actions from other
+blockchains (which is a key part of how <<chain_joins,chain joins>>
+work).
+
+
+=== Permissions
+
+Permissions are the specific actions that your application supports
+that can be assigned to roles.
+
+For example, in a voting application, the `add_vote` action which
+records a single vote on a blockchain could be assigned to the
+`can_vote` role, permitting any user with the `can_vote` role to
+dispatch the `add_vote` action.
+
+Each Interbit application includes a special permission called `*` (the
```

Comment:
The * is also a wildcard and can also be used to add bulk actions to a role. For example, if you had three actions:
* ADMIN_ADD_USER
* USER_DO_A_THING
* USER_DO_ANOTHER_THING

you could add a role with USER_* permissions that did all the user things.